src/main/java/io/fusionauth/http/server/HTTPRequest.java

```diff
@@ -264,14 +264,16 @@ public String getBaseURL() {
   }
 
   public byte[] getBodyBytes() throws BodyException {
-    if (bodyBytes == null && inputStream != null) {
-      try {
-        bodyBytes = inputStream.readAllBytes();
-      } catch (IOException e) {
-        throw new BodyException("Unable to read the HTTP request body bytes", e);
+    if (bodyBytes == null) {
```

Comment:
Previously this would return `new byte[]` on the second call causing the caller to get a different result on the second call.